dns/resolver.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dns
+
+/*
```

Comment:
I prefer to use `//` for function or type documentations, and use `/*` for package documentation (because it is longer). It's not a hard rule, just a convention.

Reply:
More info: https://google.github.io/styleguide/go/decisions#package-comments